index.js

```diff
@@ -44,9 +64,26 @@ function mongo(options) {
     log : log 
   });
 
+  let mongoPool = createMongoPool();
+
   return function* koaMongo(next) {
+
     this.mongo = yield mongoPool.acquire.bind(mongoPool);
+
     if (!this.mongo) this.throw('Fail to acquire one mongo connection');
+
+    if (!(this.mongo).serverConfig.isConnected()) {
+
+      yield wait(reconnectTries, reconnectInterval);
```

Comment:
Just wait for mongodb restart? This will block request from user for a long time.

Reply:
Releasing the resource that failed the connection test and throwing an error will still require the server app.using koa-mongo to restart in order to get a new mongo connection pool.
Also, the wait is done to give MongoClient time to perform the reconnectTries, before creating a new pool. 
We want to be able to call the generic-pool constructor when the resources acquired from the current pool fail the mongo connection test.
Another option would be to drain the pool, so that when we try to acquire a resource it throws an error and we know when to create a new pool. 

```javascript
  try {
      this.mongo = yield mongoPool.acquire.bind(mongoPool);
    } catch(e) {
      mongoPool = createMongoPool();
      this.mongo = yield mongoPool.acquire.bind(mongoPool);
    }

    if (!this.mongo) this.throw('Fail to acquire one mongo connection');

    if (!(this.mongo).serverConfig.isConnected()) {

      mongoPool.drain();
      this.throw('Cannot connect to MongoDB')
    }
```